walk/walk.go

```diff
@@ -122,16 +122,15 @@ func Walk(c *config.Config, cexts []config.Configurer, dirs []string, mode Mode,
 		log.Printf("error loading .bazelignore: %v", err)
 	}
 
-	visit(c, cexts, isBazelIgnored, knownDirectives, updateRels, wf, c.RepoRoot, "", false)
+	visit(c, cexts, isBazelIgnored, knownDirectives, updateRels, wf, []string{}, "", false)
 }
 
-func visit(c *config.Config, cexts []config.Configurer, isBazelIgnored isIgnoredFunc, knownDirectives map[string]bool, updateRels *UpdateFilter, wf WalkFunc, dir, rel string, updateParent bool) {
-	if isBazelIgnored(rel) {
-		return
-	}
-
+func visit(c *config.Config, cexts []config.Configurer, isBazelIgnored isIgnoredFunc, knownDirectives map[string]bool, updateRels *UpdateFilter, wf WalkFunc, relParts []string, rel string, updateParent bool) {
 	haveError := false
 
+	// Absolute path to the directory being visited
+	dir := path.Join(c.RepoRoot, filepath.FromSlash(rel))
```

Comment:
push this down closer to the loop in case we will early exit?

Reply:
It's used on the next line to do the `ReadDir`...